plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java

```diff
@@ -40,8 +71,114 @@ public TestResult execute(Map<String, String> testParameters) throws Exception {
     }
 
     @Override
+    @Nullable
     public ProcessHandler execute(TestExecutionContext context) {
+        String projectPath = context.getProjectPath();
+        IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectPath);
+        IJavaProject javaProject = JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+        if (javaProject.exists()) {
+            return startTestProcess(javaProject, context);
+        }
+
+        return null;
+    }
+
+    private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
+        File suiteFile = createSuite(context, javaProject);
+        if (suiteFile == null) {
+            throw new RuntimeException("Can't create TestNG suite xml file.");
+        }
+
+        JavaParameters parameters = new JavaParameters();
+        parameters.setJavaExecutable("java");
+        parameters.setMainClassName("org.testng.CheTestNGLauncher");
+        parameters.setWorkingDirectory("/tmp");
```

Comment:
Maybe we just run the corresponding TestNG tests (on an example project) in maven with `mvn test` or simply the standard build command, and compare where the test outputs are generated  ? I assume they will be generated either at the root of the project or maybe inside the `target` directory. In the last case, this would mean that an option certainly exists in testNG to specify the place of the test output directory without changing process working directory to `/tmp`.

IMO we should do that same as what maven does, because as soon as the developer will run the build command, the tests will be run in maven anyway.